packages/engine-server/src/workspace/service.ts

```diff
@@ -467,6 +468,53 @@ export class WorkspaceService implements Disposable, IWorkspaceService {
     return vault;
   }
 
+  async migrateVaultToSelfContained({ vault }: { vault: DVault }) {
+    if (vault.seed || vault.workspace) {
+      // Unsupported vaults are filtered in the commands that use this function,
+      // but also adding a sanity check here.
+      throw new DendronError({
+        message:
+          "Seed vaults and workspace vaults are not yet supported for automated migration.",
+      });
+    }
+    const newVault: SelfContainedVault = { ...vault, selfContained: true };
+
+    const vaultFolder = path.join(this.wsRoot, vault.fsPath);
+    const tmpFolder = `${vault.fsPath}-migrate-${genUUID()}`;
+    // Rename the vault temporarily, create a new folder with the same name,
+    // then move the vault into the `notes` folder inside the new folder. This
+    // is the fastest and most reliable way to move all notes and assets into
+    // the new folder, since we don't have to move each file individually.
+    await fs.move(vaultFolder, tmpFolder);
```

Comment:
does this need to be prefixed? (eg. `wsRoot/tmpFolder` or `tmpdir()/tmpfolder`?) getting this error https://www.loom.com/i/028126cbf1e942baad2387fd4e45973d

Reply:
Yup, thanks!